lib/connectors/crawler/scheduler.rb

```diff
@@ -22,11 +22,48 @@ def connector_settings
         []
       end
 
+      def when_triggered
+        loop do
+          connector_settings.each do |cs|
+            # crawler only supports :sync
+            if sync_triggered?(cs)
+              yield cs, :sync, nil
+              next
+            end
+
+            custom_schedule_triggered, schedule_key = custom_schedule_triggered?(cs)
```

Comment:
It looks like `custom_schedule_triggered` is not needed here. I would simplify it a bit

```Ruby
schedule_key = triggered_schedule_key(cs)
yield cs, :sync, schedule_key if schedule_key
```

Also, methods ending with `?` should return only boolean value they are called `boolean methods`.